From 285538e07a6dcf5a8d6bf52028e2747b6c482657 Mon Sep 17 00:00:00 2001
From: Yasufumi Ogawa <yasufum.o@gmail.com>
Date: Mon, 12 Aug 2019 14:37:49 +0900
Subject: [PATCH 3/3] docs: add usage of blacklist for SPP container

This update is to add description of usage of blacklist option. It
describes how to inspect PCI addresses and how to specify them while
launching app container.

Signed-off-by: Yasufumi Ogawa <yasufum.o@gmail.com>
---
 docs/guides/tools/sppc/getting_started.rst | 45 +++++++++++++++++++++-
 1 file changed, 44 insertions(+), 1 deletion(-)

diff --git a/docs/guides/tools/sppc/getting_started.rst b/docs/guides/tools/sppc/getting_started.rst
index d92b55f..4834cc0 100644
--- a/docs/guides/tools/sppc/getting_started.rst
+++ b/docs/guides/tools/sppc/getting_started.rst
@@ -256,9 +256,52 @@ app container launcher.
     Done
     testpmd>
 
-
 It launches ``testpmd`` in foreground mode.
 
+.. note::
+
+    DPDK app container tries to own ports on host which are shared with host
+    and containers by default. It causes a confliction between SPP running on
+    host and containers and unexpected behavior.
+
+    To avoid this situation, it is required to use ``-b`` or
+    ``--pci-blacklist`` EAL option to exclude ports on host. PCI address of
+    port can be inspected by using ``dpdk-devbind.py -s``.
+
+If you have ports on host and assign them to SPP, you should to exclude them
+from the app container by specifying PCI addresses of the ports with ``-b``
+or ``--pci-blacklist``.
+
+You can find PCI addresses from ``dpdk-devbind.py -s``.
+
+.. code-block:: console
+
+    # Check the status of the available devices.
+    dpdk-devbind --status
+    Network devices using DPDK-compatible driver
+    ============================================
+    0000:0a:00.0 '82599ES 10-Gigabit' drv=igb_uio unused=ixgbe
+    0000:0a:00.1 '82599ES 10-Gigabit' drv=igb_uio unused=ixgbe
+
+    Network devices using kernel driver
+    ===================================
+    ...
+
+In this case, you should exclude ``0000:0a:00.0`` and ``0000:0a:00.1``
+with ``-b`` option.
+
+.. code-block:: console
+
+    # Terminal 3
+    $ cd /path/to/spp/tools/sppc
+    $ app/testpmd.py -l 3-4 -d 1,2 \
+      -b 0000:0a:00.0 0000:0a:00.1
+    sudo docker run -it \
+    ...
+    -b 0000:0a:00.0 \
+    -b 0000:0a:00.1 \
+    ...
+
 
 .. _sppc_gs_run_apps:
 
-- 
2.17.1

